src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt

```diff
@@ -205,16 +207,42 @@ class ADRController(private val encryption: Encryption,
                 listOf("name" to resourceFileName, "description" to artefact.second, "hash" to file.md5sum(),
                         "resource_type" to resourceType)
         val adr = adrClientBuilder.build()
+        val newDatasetHash = commonParameters[2].second
```

Comment:
This is a little bit obfuscated, could you just set `val newDatasetHash = file.md5sum()` before constructing commonParameters, then use that value when making the list.